src/Spice86.Shared/Utils/ConvertUtils.cs

```diff
@@ -99,12 +102,36 @@ public static uint BytesToInt32(byte[] data, int start) {
     public static byte[] HexToByteArray(string valueString) {
         byte[] res = new byte[valueString.Length / 2];
         for (int i = 0; i < valueString.Length; i += 2) {
-            string hex = valueString.Substring(i, 2);
-            res[i / 2] = byte.Parse(hex, NumberStyles.HexNumber);
+            string hex = valueString.Substring(i, HexadecimalByteDigitLength);
+            res[i / HexadecimalByteDigitLength] = byte.Parse(hex, NumberStyles.HexNumber);
         }
 
         return res;
     }
+    
+    /// <summary>
+    /// Tries to convert a hexadecimal string to a byte array.
+    /// </summary>
+    /// <param name="valueString">The hexadecimal string to convert.</param>
+    /// <param name="bytes">The byte array representation of the hexadecimal string.</param>
+    /// <returns><c>False</c> if the conversion fails, <c>True</c> otherwise</returns>
+    public static bool TryParseHexToByteArray(string valueString, [NotNullWhen(true)] out byte[]? bytes) {
```

Comment:
How is this different from [Convert.FromHexString](https://learn.microsoft.com/en-us/dotnet/api/system.convert.fromhexstring?view=net-8.0)?

Reply:
It's used to prevent an exception if the UI input from the user is not recognized.